src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public boolean add(T value) {
-        return false;
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    public MyLinkedList() {
+    }
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(Node<T> previous, T data, Node<T> next) {
+            this.data = data;
+            this.next = next;
+            this.previous = previous;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public boolean add(T data) {
+        insertLast(data);
+        return true;
+    }
 
+    @Override
+    public void add(T data, int index) {
+        checkBounds(index);
+        if (index == size) {
+            insertLast(data);
+        } else {
+            insertBefore(data, getNode(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        T[] array = (T[])list.toArray();
+        for (int k = 0; k < array.length; k++) {
+            insertLast(array[k]);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return getNode(index).data;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = getNode(index);
+        T oldValue = node.data;
+        node.data = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return removing(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
```

Comment:
let's simplify this method, use on loop in the method independently if passed value is null or non null.

Reply:
fixed!